examples/algorithms/plot_metric_closure.py

```diff
@@ -0,0 +1,65 @@
+"""
+==============
+Metric Closure
+==============
+
+The *metric closure* of a graph is the complete graph in which
+each edge weight corresponds to the shortest-path distance
+between two nodes in the original graph.
+
+This example demonstrates the *unweighted* version of the metric closure,
+constructed using :func:`networkx.all_pairs_shortest_path_length`.
+Here, each edge is assumed to have weight 1, so distances correspond to
+the minimum number of hops between nodes.
+
+For weighted graphs, the metric closure can be obtained instead with
+:func:`networkx.all_pairs_dijkstra_path_length`, which computes distances
+using the minimum total edge weights along paths.
+"""
+
+import networkx as nx
+import matplotlib.pyplot as plt
+from pprint import pprint
+
+# Creating five nodes from index 0 to 4
+G = nx.house_graph()
+
+# Find the all-pairs shortest distance
+# all_pairs_shortest_path_length returns a one-time generator,
+# so wrap with dict() to materialize results for reuse.
+paths = dict(nx.all_pairs_shortest_path_length(G))
+
+
+# Dictionary format: u: {v: d}, meaning "distance d from node u to node v"
+pprint(paths)
+
+# Metric closure of G
+M = nx.Graph()
+for u, dist_dict in paths.items():
+    for v, d in dist_dict.items():
+        if u != v:  # avoid self-loops
+            M.add_edge(u, v, distance=d)
+# Visualize the G and metric closure of G
+fig, axes = plt.subplots(1, 2, figsize=(10, 4))
+
+pos = nx.circular_layout(G)
```

Comment:
I'm not sure this is what you want, but it is a standard set of positions for the "house graph" (obtained from the [house_graph gallery example](https://networkx.org/documentation/stable/auto_examples/drawing/plot_house_with_colors.html)). These position supposedly make it look like a house.
```suggestion
pos = {0: (0, 0), 1: (1, 0), 2: (0, 1), 3: (1, 1), 4: (0.5, 2.0)}
```

Reply:
Oops - this was my fault. The house_graph instance `G` should indeed be laid out like this, or using something like `spring_layout` will give a good result (as @Aka2210 had originally done).

I meant to suggest the `circular_layout` *only* for the metric closure graph, since it's a complete graph and circular layouts tend to work well for those!